code/game/objects/items/mail.dm

```diff
@@ -332,3 +353,166 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/datum/mind/madeby
```

Comment:
This is still a hard-del if the mind destroys.

I don't think you need to hold onto this at all. If you log when a mail bomb is created, and you log the ckey of the person who made the bomb, then it is trivial for admins to solve.

Reply:
Isn't it how it works in my case? On bomb creation ckey and name being cached, so any hard deletion of owner - logs still will be accurate.

Reply:
A hard deletion is when you are holding onto a reference to something that gets deleted. These are always bugs. You would need to handle that, but I don't think you should bother and instead should not hold onto the mind at all.

Reply:
ah, i see... Weak Ref can fix the situation?

Reply:
Yes but like I said I really don't think it's necessary